desktop-widgets/divelogexportdialog.cpp

```diff
@@ -275,31 +274,77 @@ void DiveLogExportDialog::export_TeX(const char *filename, const bool selected_o
 
 		QString star = "*";
 		QString viz = star.repeated(dive->visibility);
-		int i;
+		QString rating = star.repeated(dive->rating);
 
-		for (i = 0; i < MAX_CYLINDERS; i++)
-			if (is_cylinder_used(dive, i))
-				delta_p.mbar += dive->cylinder[i].start.mbar - dive->cylinder[i].end.mbar;
+		int i;
+		int qty_cyl;
+		int qty_weight;
+		float total_weight;
 
 		if (need_pagebreak)
 			put_format(&buf, "\\vfill\\eject\n");
 		need_pagebreak = true;
 		put_format(&buf, "\\def\\date{%04u-%02u-%02u}\n",
 		      tm.tm_year, tm.tm_mon+1, tm.tm_mday);
 		put_format(&buf, "\\def\\number{%d}\n", dive->number);
-		put_format(&buf, "\\def\\place{%s}\n", site ? site->name : "");
-		put_format(&buf, "\\def\\spot{}\n");
+		put_format(&buf, "\\def\\sitename{%s}\n", site ? site->name : "");
+		put_format(&buf, "\\def\\gpslat{%f}\n", site ? site->latitude.udeg / 1000000.0 : 0);
+		put_format(&buf, "\\def\\gpslon{%f}\n", site ? site->longitude.udeg / 1000000.0 : 0);
+		put_format(&buf, "\\def\\computer{%s}\n", dive->dc.model);
 		put_format(&buf, "\\def\\country{%s}\n", country.toUtf8().data());
-		put_format(&buf, "\\def\\entrance{}\n");
 		put_format(&buf, "\\def\\time{%u:%02u}\n", FRACTION(dive->duration.seconds, 60));
-		put_format(&buf, "\\def\\depth{%u.%01um}\n", FRACTION(dive->maxdepth.mm / 100, 10));
-		put_format(&buf, "\\def\\gasuse{%u.%01ubar}\n", FRACTION(delta_p.mbar / 100, 10));
+		put_format(&buf, "\\def\\maxtemp{%.2f}\n", (dive->maxtemp.mkelvin) ? (dive->maxtemp.mkelvin / 1000.0) - 273.15 : 0);
+		put_format(&buf, "\\def\\mintemp{%.2f}\n", (dive->mintemp.mkelvin) ? (dive->mintemp.mkelvin / 1000.0) - 273.15 : 0);
+		put_format(&buf, "\\def\\watertemp{%.2f}\n", (dive->watertemp.mkelvin) ? (dive->watertemp.mkelvin / 1000.0) - 273.15 : 0);
+		put_format(&buf, "\\def\\airtemp{%.2f}\n", (dive->airtemp.mkelvin) ? (dive->airtemp.mkelvin / 1000.0) - 273.15 : 0);
+		put_format(&buf, "\\def\\maxdepth{%u.%01u m}\n", FRACTION(dive->maxdepth.mm / 100, 10));
+		put_format(&buf, "\\def\\avedepth{%u.%01u m}\n", FRACTION(dive->meandepth.mm / 100, 10));
+
+		//Code block prints start/end press. for all cylinders used, number of cyl. used, SAC, and total delta_p
+		qty_cyl = 0;
+		for (i = 0; i < MAX_CYLINDERS; i++)
+			if (is_cylinder_used(dive, i)){
+				put_format(&buf, "\\def\\cyl%ddescription{%s}\n", i + 1, dive->cylinder[i].type.description);
```

Comment:
As embarrassed as I am to admit it, you're right!  This doesn't compile - I should have had a \% rather than just % in the output.  I'll get that fixed.  However, once I inserted that backslash and checked the template for untranslated keywords (\klein = \small), it processed fine - though it's not pretty.

There is one change in that I renamed /place as /sitename. (The template would need that one line to be updated.)

Here is a sample output file produced by this code:
`\def\cyl1description{AL80}
\def\cyl1mixO2{21%}  \\note should read "...{21\%}" in fixed version.  Same for all fields below.
\def\cyl1mixN2{79%}
\def\cyl1mixHe{0%}
\def\cyl1startpress{213.7 bar}
\def\cyl1endpress{34.4 bar}
\def\cyl2description{AL100}
\def\cyl2mixO2{40%}
\def\cyl2mixN2{0%}
\def\cyl2mixHe{60%}
\def\cyl2startpress{200.0 bar}
\def\cyl2endpress{20.0 bar}
\def\qtycyl{2}`

So far, I have been diving single tanks exclusively, so all I really need to print is the data for cyl1.

I'll take a look at the case where cylinder #3 is used, but cylinder #2 is not, and see if I can get to work out nicely.  For my implementation, I want to be able to print the cylinders that were used, and I don't care about the ones that were not.